repositories/spacedOrderStrategy.js

```diff
@@ -0,0 +1,76 @@
+const { map } = require("lodash");
+
+const SPACING = 1000;
+
+const calculateMidPoint = (orderBefore, orderAfter) =>
+  Math.round((orderBefore + orderAfter) / 2);
+
+const valuesHaveConverged = (orderBefore, orderAfter) =>
+  Math.abs(orderAfter - orderBefore) < 2;
+
+const getNextOrderValue = (trx, SectionId) => {
+  return trx("Pages")
+    .column(trx.raw(`COALESCE(max("order"), 0) + ${SPACING} as "order"`))
+    .where({ SectionId, isDeleted: false });
+};
+
+const setOrder = (trx, id, SectionId, order) => {
+  return trx("Pages")
+    .where({ id })
+    .update({ order, SectionId })
+    .returning("*");
+};
+
+const getAdjacentRows = (trx, sectionId, position) => {
+  return trx("Pages")
+    .columns("id", "order")
+    .where({
+      SectionId: sectionId,
+      isDeleted: false
+    })
+    .orderBy("order")
+    .offset(Math.max(0, position - 1))
+    .limit(2)
+    .then(result => map(result, "order"));
+};
+
+const makeSpaceAfter = (trx, order, SectionId) => {
+  return trx("Pages")
+    .where({ SectionId })
+    .andWhere("order", ">", order)
+    .increment("order", SPACING);
+};
+
+const movePage = async (trx, { id, sectionId, position }) => {
+  position = Math.max(0, position);
+
+  let rows = await getAdjacentRows(trx, sectionId, position);
+
+  // position > number of rows, then we need to get max current value
```

Comment:
We just chatted about this function and decided it was a little confusing that the 3 special cases all always run and override `rows`. So the `// inserting at start of list` condition will always take precedence.

It's a little unclear what the intended code paths are at first... or fourth glance 😄